content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md

```diff
@@ -0,0 +1,487 @@
+---
+title: Troubleshooting the Target Allocator
+description: Tips for troubleshooting the Target Allocator
+cSpell:ignore: bleh targetallocator
+---
+
+If you’ve enabled
+[Target Allocator](/docs/kubernetes/operator/target-allocator/) service
+discovery on the [OpenTelemetry Operator](/docs/kubernetes/operator), and your
+metrics aren’t getting scraped, then there are a few troubleshooting steps that
+you can take to help you understand what’s going on and to get things back on
+track.
+
+## Troubleshooting Steps
+
+### 1- Did you deploy all of your resources to Kubernetes?
+
+Although this may seem obvious and straightforward, it can happen! First things
+first: make sure that you have deployed all relevant resources to your
+Kubernetes cluster.
+
+### 2- Do you know if metrics are actually being scraped?
+
+After you’ve deployed all of your resources to Kubernetes, check to make sure
+that the Target Allocator is actually scraping metrics from your
+[`ServiceMonitor`](https://observability.thomasriley.co.uk/prometheus/configuring-prometheus/using-service-monitors/#:~:text=The%20ServiceMonitor%20is%20used%20to,build%20the%20required%20Prometheus%20configuration.)(s)
+and/or
+[`PodMonitor`](https://prometheus-operator.dev/docs/user-guides/getting-started/#using-podmonitors)(s).
+
+Suppose that you have this `ServiceMonitor` definition:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: sm-example
+  namespace: opentelemetry
+  labels:
+    app.kubernetes.io/name: py-prometheus-app
+    release: prometheus
+spec:
+  selector:
+    matchLabels:
+      app: my-app
+  namespaceSelector:
+    matchNames:
+      - opentelemetry
+  endpoints:
+    - port: prom
+      path: /metrics
+    - port: py-client-port
+      interval: 15s
+    - port: py-server-port
+```
+
+this `Service` definition:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: py-prometheus-app
+  namespace: opentelemetry
+  labels:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+spec:
+  selector:
+    app: my-app
+    app.kubernetes.io/name: py-prometheus-app
+  ports:
+    - name: prom
+      port: 8080
+```
+
+and this `OpenTelemetryCollector` definition:
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: opentelemetry
+spec:
+  mode: statefulset
+  image: otel/opentelemetry-collector-contrib:0.102.1
```

Comment:
We might consider auto-updating this collector version here, but this can happen in a later update

Reply:
@svrnm I can also leave it out, and then the OTel Operator will just use whatever the default image is

Reply:
if that works, I would prefer that, except someone from @open-telemetry/operator-approvers disagrees

Reply:
I'll hold off on updating until we get feedback from @open-telemetry/operator-approvers, then. 🙂

Reply:
It's better to leave it blank in examples, as a rule. We do run the risk of our examples being broken by future releases, but in the case of K8s API Versioning, breaking changes can only happen with an explicit version change, the worst that can happen is that the whole resource will be rejected as unsupported. There won't be subtle breakage.

Reply:
Noted, and removed!